pyworkflow/template.py

```diff
@@ -209,3 +238,36 @@ def sortListByPluginName(self):
 
         return self
 
+    def addScipionTemplates(self, tempId=None):
+        # Check if there is any .json.template in the template folder
+        # get the template folder (we only want it to be included once)
+        templateFolder = Config.getExternalJsonTemplates()
+        for templateName in glob.glob1(templateFolder,
+                                       "*" + SCIPION_JSON_TEMPLATES):
+            t = Template("local", os.path.join(templateFolder, templateName))
+            if tempId is not None:
+                if t.getObjId() == tempId:
+                    self.addTemplate(t)
+                    break
+            else:
+                self.addTemplate(t)
+
+    def addPluginTemplates(self, tempId=None):
+        """
+        Get the templates provided by all plugins.
+        :return: a list of templates
+        """
+        # Check if other plugins have json.templates
+        domain = Config.getDomain()
+        # Check if there is any .json.template in the template folder
+        # get the template folder (we only want it to be included once)
+        for pluginName, pluginModule in domain.getPlugins().items():
+            tempListPlugin = pluginModule.Plugin.getTemplates()
+            for t in tempListPlugin:
+                if tempId is not None:
+                    if t.getObjId() == tempId:
+                        self.addTemplate(t)
+                        break
+                else:
+                    self.addTemplate(t)
+
```

Comment:
Really nice job, I think it's a very good functionality having the option of also calling one template using its id, apart from the previous existing one introducing the whole path of the file.